roadmap/implementers-guide/src/node/disputes/dispute-coordinator.md

```diff
@@ -4,10 +4,324 @@ The coordinator is the central subsystem of the node-side components which parti
+import approval votes at a later point in time as well (in particular we need to
+make sure the dispute can conclude, but more on that later).
+
+As mentioned already previously, importing votes is most efficient when batched.
+At the same time approval voting and disputes are running concurrently so
+approval votes are expected to trickle in still, when a dispute is already
+ongoing.
+
+Hence, we have the following requirements for importing approval votes:
+
+1. Only import them when there is a dispute, because otherwise we are
+   wasting lots of resources _always_ for the exceptional case of a dispute.
+2. Import votes batched when possible, to avoid quadratic import complexity.
+3. Take into account that approval voting is still ongoing, while a dispute is
+   already running.
+
+With a design where approval voting sends votes to the dispute-coordinator by
+itself, we would need to make approval voting aware of ongoing disputes and
+once it is aware it could start sending all already existing votes batched and
+trickling in votes as they come. The problem with this is, that it adds some
+unnecessary complexity to approval voting and also we might still import most of
+the votes unbatched, but one-by-one, depending on what point in time the dispute
+was raised.
+
+Instead of the dispute coordinator telling approval-voting that a dispute is
+ongoing for approval-voting to start sending votes to the dispute coordinator,
+it would make more sense if the dispute-coordinator would just ask
+approval-voting for votes of candidates that are currently disputed. This way
+the dispute-coordinator can also pick the time when to ask and we can therefore
+maximize the amount of batching.
+
+Now the question remains, when should the dispute coordinator ask
+approval-voting for votes? As argued above already, querying approval votes at
+the beginning of the dispute, will likely already take care of most malicious
+votes. Still we would like to have a record of all, if possible. So what are
+other points in time we might query approval votes?
+
+In fact for slashing it is only relevant to have them once the dispute
+concluded, so we can query approval voting the moment the dispute concludes!
+There are two potential caveats with this though:
+
+1. Timing: We would like to rely as little as possible on implementation details
+   of approval voting. In particular, if the dispute is ongoing for a long time,
+   do we have any guarantees that approval votes are kept around long enough by
+   approval voting? So will approval votes still be present by the time the
+   dispute concludes in all cases? The answer should luckily be yes: As long as
+   the chain is not finalized, which has to be the case once we have an ongoing
+   dispute, approval votes have to be kept around (and distributed) otherwise we
+   might not be able to finalize in case the validator set changes for example.
+   Conclusively we can rely on approval votes to be still available when the
```

Comment:
I addressed this above - only guaranteed to hold true for some set of f+1 honest validators.